src/fheroes2/agg/agg_image.cpp

```diff
@@ -1907,6 +1910,56 @@ namespace fheroes2
 
                 break;
             }
+            case ICN::BUTTON_HSCORES_VERTICAL_CAMPAIGN:
+            case ICN::BUTTON_HSCORES_VERTICAL_EXIT:
+            case ICN::BUTTON_HSCORES_VERTICAL_STANDARD: {
+                _icnVsSprite[id].resize( 2 );
+
+                const int32_t originalID = ICN::HISCORE;
+                int32_t originalICNIndex = 0;
```

Comment:
These two variables are used in `GetICN( int icnId, uint32_t index )`. So to avoid type conversions we can declare these variables using the corresponding types:
```suggestion
                const int originalID = ICN::HISCORE;
                uint32_t originalICNIndex = 0;
```